src/main/java/com/google/sps/servlets/CreateEventServlet.java

```diff
@@ -21,37 +22,44 @@
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        long epoch;     // Hold date's convertion to epoch (value to be stored in Datastore).
         Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
         KeyFactory keyFactory = datastore.newKeyFactory().setKind("event");
         double price = Double.parseDouble(request.getParameter("price"));
-        long date = Long.parseLong(request.getParameter("date"));
         String address = Jsoup.clean(request.getParameter("address"), Whitelist.none()),
+            date = Jsoup.clean(request.getParameter("date"), Whitelist.none()),
             state = Jsoup.clean(request.getParameter("state"), Whitelist.none()),
             type = Jsoup.clean(request.getParameter("type"), Whitelist.none()),
-            details = Jsoup.clean(request.getParameter("description"), Whitelist.none()),
+            description = Jsoup.clean(request.getParameter("description"), Whitelist.none()),
             zipCode = Jsoup.clean(request.getParameter("zip"), Whitelist.none());
 
-        /**
-         * TODO (Josh-hdz): posible validations (ex. one event per address)
-         */
-
-        FullEntity taskEntity =
-            Entity.newBuilder(keyFactory.newKey())
-                .set("type", type)
-                .set("date", date)
-                .set("details", details)
-                .set("price", price)
-                .set("address", address)
-                .set("state", state)
-                .set("zip", zipCode)
-                .build();
-        datastore.put(taskEntity);
-
-        response.sendRedirect("/index.html");
-        /**
-         * TODO (Josh-hdz): add response to let know front-end wether or not the
-         *                  event was added to database
-         */
-        
+        try{
+            epoch = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm").parse(date).getTime() / 1000;
+
+            /**
+             * TODO (Josh-hdz): posible validations (ex. one event per address)
+             */
+
+            FullEntity taskEntity =
+                Entity.newBuilder(keyFactory.newKey())
+                    .set("type", type)
+                    .set("epoch", epoch)
+                    .set("description", description)
+                    .set("price", price)
+                    .set("address", address)
+                    .set("state", state)
+                    .set("zip", zipCode)
+                    .build();
+            datastore.put(taskEntity);
+
+            response.sendRedirect("/index.html");
+            /**
+             * TODO (Josh-hdz): add response to let know front-end wether or not the
+             *                  event was added to database
+             */
+
+        } catch (java.text.ParseException e) {
+            response.sendError(1, "Invalid date format");
```

Comment:
Is it possible for the user to enter an invalid date, considering we have a date picker? If not, I would add a comment to the `catch` saying that the catch is unlikely to run because the date picker should send valid input as `date`.

I currently see this try/catch and see that if the user doesn't have a correct date, then an event is not created. I think that makes sense, since the date is required for the event, but since we currently silently fail and don't notify the user, having a comment in the code could help reassure a code reader that in most cases an event will actually be created.

Reply:
I added the try/catch because java will not compile there is't a try/catch. I'll add the comment about it.